### ISSUE TITLE
Feature: SCEP/NDES cert enrollment support

### ISSUE THREAD

**Issue by josh-hemphill** [caddyserver/caddy#4020]
Now that [server identity management](https://github.com/caddyserver/caddy/pull/3994) has landed in the beta for [2.4](https://github.com/caddyserver/caddy/releases/tag/v2.4.0-beta.1), and local PKI module has been around a while, It would be awesome to get SCEP/NDES support for local-domain-only certs especially for distributing certs for server-to-server trust.

#3021 Was the last discussion on the topic

It looks like some work was done on a Go SCEP cliet/server https://github.com/micromdm/scep

MS Docs: https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-r2-and-2012/hh831498(v=ws.11)

**Comment by hslatman** [comment]
Recently I started a Caddy module for supporting SCEP: https://github.com/hslatman/caddy-scep. It's really rudimentary and it doesn't work fully out of the box.

Since that, I've also forked smallstep/certificates and worked on integrating SCEP there. Progress is here: https://github.com/smallstep/certificates/issues/14. It may be possible to expose the SCEP support in Smallstep (if/when merged) to Caddy.
